### ISSUE TITLE
Nheko crashes on startup when passing a matrix URI, emptying the cache

### ISSUE THREAD

**Issue by MayeulC** [Nheko-Reborn/nheko#401]
### Describe the bug
I've been meaning to try #370. After reading the code, I figured I could pass a matrix URI as a parameter while nheko was running.

### To Reproduce
1. Start nheko
2. Start nheko passing a URI of a room you are in, for instance `io.github.NhekoReborn.Nheko "matrix:room/pinephone:matrix.org"
3. Observe the log
4. Restart nheko: you are logged out, encryption keys have disappeared

### Expected behavior
Either open the room, or refuse to open it (I haven't double-checked that the URI is valid).

### System:

- Nheko version: nheko 0.8.0-a9faece 
- Installation method: flathub
- Operating System: Arch Linux
- Desktop Environment: sway

### Logs

When trying to open a link:
```
% io.github.NhekoReborn.Nheko "matrix:room/pinephone:matrix.org"
[2021-01-22 11:02:48.031] [ui] [info] WebRTC: initialised GStreamer 1.18.3
[2021-01-22 11:02:48.431] [db] [critical] failure during boot: mdb_txn_begin: Ressource temporairement non disponible
[2021-01-22 11:02:48.433] [db] [info] deleted cache files from disk
[2021-01-22 11:02:48.434] [net] [info] falling back to initial sync
[2021-01-22 11:02:48.434] [crypto] [info] creating new olm account
```

When restarting nheko:

```
% io.github.NhekoReborn.Nheko                                                                                                                                                                   :)
[2021-01-22 11:03:09.441] [ui] [info] WebRTC: initialised GStreamer 1.18.3
[2021-01-22 11:03:09.823] [db] [info] initializing LMDB
[2021-01-22 11:03:09.824] [crypto] [info] creating new olm account
[2021-01-22 11:03:09.824] [crypto] [info] ed25519   : d/18Z6vv7Ar+ztqlv71MSewWOr8ooj7sbfR4jABGFnQ
[2021-01-22 11:03:09.824] [crypto] [info] curve25519: Xw/BK/NmrAri+w4QdyHBitcZbg7qJ9SlR/LQ2gQd4mM
[2021-01-22 11:03:09.824] [crypto] [info] generating one time keys
[2021-01-22 11:03:09.836] [ui] [info] jdenticon plugin not found.
[2021-01-22 11:03:09.836] [ui] [info] timelineWidth: -166
[2021-01-22 11:03:09.836] [ui] [info] timelineWidth: 18446744073709551450, min 490
[2021-01-22 11:03:09.864] [ui] [info] timelineWidth: 668
[2021-01-22 11:03:09.864] [ui] [info] timelineWidth: 668, min 490
[2021-01-22 11:03:09.865] [ui] [info] starting nheko 0.8.0-a9faece
[2021-01-22 11:03:09.913] [ui] [info] timelineWidth: 1654
[2021-01-22 11:03:09.913] [ui] [info] timelineWidth: 1654, min 490
[2021-01-22 11:03:11.815] [net] [info] TURN server(s) retrieved from homeserver:
[2021-01-22 11:03:11.815] [net] [info] username: 1611313391:@mayeul:mayeul.net
[2021-01-22 11:03:11.815] [net] [info] ttl: 3600 seconds
[2021-01-22 11:03:11.815] [net] [info] uri: turn:mayeul.net:5349?transport=udp
[2021-01-22 11:03:11.815] [net] [info] uri: turn:mayeul.net:5349?transport=tcp
[2021-01-22 11:03:15.992] [crypto] [critical] failed to upload one time keys: One time key signed_curve25519:AAAACQ already exists. Old key: {"key":"AbltREDACTED","signatures":{"@mayeul:mayeul.net":{"ed25519:nheko039-2":"IUD5REDACTED"}}}; new key: {'key': '7RUCREDACTED', 'signatures': {'@mayeul:mayeul.net': {'ed25519:nheko039-2': 'kJwxREDACTED'}}} 400
[2021-01-22 11:03:15.992] [ui] [info] dropping to the login page: Échec de la configuration des clés de chiffrement. Réponse du serveur : One time key signed_curve25519:AAAACQ already exists. Old key: {"key":"AbltREDACTED","signatures":{"@mayeul:mayeul.net":{"ed25519:nheko039-2":"IUD5REDACTED"}}}; new key: {'key': '7RUCREDACTED', 'signatures': {'@mayeul:mayeul.net': {'ed25519:nheko039-2': 'kJwxREDACTED'}}} 400. Veuillez réessayer plus tard.
[2021-01-22 11:03:15.994] [qml] [warning] qrc:/qml/MessageInput.qml:231: TypeError: Cannot read property 'input' of null (qrc:/qml/MessageInput.qml:231, )
[2021-01-22 11:03:15.996] [db] [info] deleted cache files from disk
```
I suspect there are two distinct but compounding bugs above: one with the URI scheme, and one with key uploading.

(I can translate the French bits if you want).

### Debugger backtrace

TODO :(

**Comment by deepbluev7** [comment]
Probably a bug with flatpak and the single application stuff, need to try that.

**Comment by deepbluev7** [comment]
Fixed, you need flatpak 1.11.1 at least though.
